summarization/extractor_output_preprocessor.py

```diff
@@ -87,6 +90,28 @@ def _split_content(self):
             elif content.content_type != ContentType.UNKNOWN:
                 self.embedded_content_list.append(content)
 
+    def _set_sentence_objects_list_for_title_sentences(self):
+        ''' sets the list of title sentences objects'''
+        # initialize an empty list
+        self.title_text_objects_list = list()
+
+        # collect and retrieve all embeddings at once
+        embeddings = self.sentence_embedding_model.encode([
+            text.text_string for text in self.title_text_content_list
+        ])
+
+        # instantiate and append the sentence object
+        for title_text, embedding in zip(
+                self.title_text_content_list, embeddings):
+            self.title_text_objects_list.append(
+                SentenceWithAttributes(
+                    title_text.text_string,
+                    title_text.content_index,
+                    None,
+                    embedding
+                )
+            )
```

Comment:
It looks like we are doing almost the same processing for title text and summary text. If the processing steps are not too different, would it be possible to avoid code repetition?

Reply:
Ah yes - that sounds right.
I guess it seems repeated due to the following reasons:
1) we calculate embeddings in both
2) we instantiate SentenceWithAttributes objects in both
But, the embedding is used for the SentenceWithAttributes differently:
* in instantiating them for title sentences we can do so directly
* in instantiating them for normal text we need to gather other info such as the index and the font style
so we could just separate out the method for instantiation maybe - but besides that i feel it does not have any intersection in code that can be separated 
Would be great to hear your thoughts on this!

Reply:
That makes sense! If it is so little code that they actually share, then it is better to leave it as is.